invenio_rdm_records/resources/serializers/schemaorg/schema.py

```diff
@@ -0,0 +1,343 @@
+                    )
+
+            return serialized_affiliation
+
+        affiliations = obj.get("affiliations", [])
+        return [format_affiliation(a) for a in affiliations] or missing
+
+
+class SchemaorgSchema(BaseSerializerSchema, CommonFieldsMixin):
+    """Schemaorg Marshmallow Schema."""
+
+    context = fields.Constant("http://schema.org", data_key="@context")
+    id_ = fields.Method("get_id", data_key="@id")
+    type_ = fields.Method("get_type", data_key="@type")
+    identifier = fields.Method("get_identifiers")
+    name = SanitizedUnicode(attribute="metadata.title")
+    author = fields.List(
+        fields.Nested(PersonOrOrgSchema), attribute="metadata.creators"
+    )
+    editor = fields.List(
+        fields.Nested(PersonOrOrgSchema), attribute="metadata.contributors"
+    )
+    publisher = fields.Method("get_publisher")
+    keywords = fields.Method("get_keywords")
+    datePublished = fields.Method("get_publication_date")
+    dateModified = fields.Method("get_modification_date")
+    inLanguage = fields.Method("get_language")
+    contentSize = fields.Method("get_size")
+    encodingFormat = fields.Method("get_format")
+    version = SanitizedUnicode(attribute="metadata.version")
+    license = fields.Method("get_license")
+    description = SanitizedUnicode(attribute="metadata.description")
+    # spatialCoverage = fields.Method("get_spatial_coverage")
+    funding = fields.Method("get_funding")
+
+    def get_id(self, obj):
+        """Get id. Use the DOI expressed as a URL."""
+        doi = py_.get(obj, "pids.doi.identifier")
+        if doi:
+            return doi_as_url(doi)
+        return missing
+
+    def get_type(self, obj):
+        """Get type. Use the vocabulary service to get the schema.org type."""
+        props = get_vocabulary_props(
+            "resourcetypes",
+            ["props.schema.org"],
+            py_.get(obj, "metadata.resource_type.id"),
+        )
+        return props.get("schema.org", "CreativeWork")
+
+    def get_size(self, obj):
+        """Get size."""
+        ret = None
+        size = py_.get(obj, "files.total_bytes")
+        if size:
+            ret = convert_size(size)
+        return ret or missing
+
+    def get_format(self, obj):
+        """Get format."""
+        formats = py_.get(obj, "metadata.formats")
+        return unwrap(wrap(formats)) or missing
+
+    def get_publication_date(self, obj):
+        """Get publication date."""
+        try:
+            parsed_date = parse_edtf(py_.get(obj, "metadata.publication_date"))
+        except ParseException:
+            return missing
+
+        # if date is an interval, use the start date
+        if (type(parsed_date).__name__) == "Interval":
+            parsed_date = parsed_date.lower
+        return str(parsed_date)
+
+    def get_modification_date(self, obj):
+        """Get modification date."""
+        # TODO: implement with the help of test data
+        return missing
```

Comment:
minor: isn't this just `obj["updated"]`?

Reply:
you're right, thanks!